target_s3_csv/s3.py

```diff
@@ -27,10 +27,12 @@ def log_backoff_attempt(details):
 def setup_aws_client(config):
     aws_access_key_id = config['aws_access_key_id']
     aws_secret_access_key = config['aws_secret_access_key']
+    aws_session_token = config.get('aws_session_token', None)
 
     LOGGER.info("Attempting to create AWS session")
     boto3.setup_default_session(aws_access_key_id=aws_access_key_id,
```

Comment:
Can we improve it a bit more to work the same way as pipelinewise-target-snowflake [here](https://github.com/transferwise/pipelinewise-target-snowflake/blob/a26fa575cd24bdf63603d845988db90fbc0c5229/target_snowflake/db_sync.py#L291) ?

That would add secrets, env vars, session tokens and IAM roles in one go. Creating the boto3 client object without acces/secret keys will enable boto3 to look for all possible credentials and it'd be something like:

```
        self.s3_client = boto3.client(
            's3',
            aws_access_key_id=self.connection_config.get('aws_access_key_id'),
            aws_secret_access_key=self.connection_config.get('aws_secret_access_key'),
            aws_session_token=self.connection_config.get('aws_session_token')
        )
```
f the key is not in the config then self.connection_config.get('...') will return None and boto3 will look for the next possible credential source automatically.

What do you think, does that make sense?